### PR TITLE
Extract PopoverComponent from ActionMenu

### PR DESCRIPTION
**Pull request by dgalarza**
A popover can be useful in other contexts than just the action menu. This breaks the popover out of the ActionMenuComponent and uses composition.

### CODE REVIEW

File: app/components/anchor/popover_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<%= tag.div(
+  class: class_names(
+    "absolute -left-full -top-full m-0 px-0 py-1 bg-white rounded border border-subdued popover-open:block",
```
Comment (Simonpedro) [inline app/components/anchor/popover_component.html.erb:3]: What you think of we moving the responsability of defining the padding to the consumer? The action menu would pass this padding classes while other components, like the filter in test results, would use `p-4`.

**Review by Simonpedro — COMMENTED** [review]
Left a couple of thoughts, looking good!

File: app/components/anchor/popover_component.html.erb (hunk 1 above)
Comment (dgalarza) [reply app/components/anchor/popover_component.html.erb:3]: Good call!

File: app/components/anchor/action_menu_component.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ module Anchor
   class ActionMenuComponent < Component
     renders_one :show_button, lambda { |**kwargs|
       ButtonComponent.new(
-        data: { action_menu_target: "button", testid: "action-menu-btn" },
+        data: { popover_target: "button", testid: "action-menu-btn" },
```
[the remark this answers is not in the document]
Comment (dgalarza) [reply app/components/anchor/action_menu_component.rb]: I like this idea, I went with a view helper as that felt more semantic within view components as well as easily reusable in other places that you want to spin up a popover. Let me know what you think.

**Review by Simonpedro — APPROVED** [review]
🎉

File: app/components/anchor/action_menu_component.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Simonpedro) [reply app/components/anchor/action_menu_component.rb]: I like it!

**Pull request merged**